lib/perfdata/elasticsearchwriter.hpp

```diff
@@ -37,6 +40,7 @@ class ElasticsearchWriter final : public ObjectImpl<ElasticsearchWriter>
 	std::mutex m_DataBufferMutex;
 
 	void AddCheckResult(const Dictionary::Ptr& fields, const Checkable::Ptr& checkable, const CheckResult::Ptr& cr);
+	void AddTemplateTags(const Dictionary::Ptr& fields, const Host::Ptr& host, const Service::Ptr& service, const CheckResult::Ptr& cr);
```

Comment:
That is true and I also thought about doing so.
My reasoning was the following:
For the Resolver inside of the method I need to get the Host::Ptr and Service::Ptr from the Checkable::Ptr like it is done in other places.
But at all sources where AddTemplateTags will be called from this Host and Service retrieval has been done already just a couple of lines above.
So I would do the same "conversion" multiple times and therefore opted to call the Method with Host::Ptr and Service::Ptr directly.
If you would prefer the Checkable::Ptr way I can also implement it that way.

Reply:
> If you would prefer the Checkable::Ptr way I can also implement it that way.

The thing is, currently it looks like as the service parameter is always required and implies that it may never be a `nullptr`, thus I would either use `Checkable::ptr` instead and do the one-line casting again (my preferred variant) or make the `service` parameter optional.